lib/pokegem.rb

```diff
@@ -1,24 +1,126 @@
-require "pokegem/version"
+require 'pokegem/version'
 require 'typhoeus'
 require 'json'
+require 'uri'
 
 module Pokegem
-  class << self
-    BASE_URL  = 'http://pokeapi.co/api/v1'
-    RESOURCES = %w(pokedex pokemon move ability type egg description sprite game)
+  class Cache
+    def initialize
+      @memory = {}
+    end
+
+    def get(request)
+      @memory[request]
+    end
 
-    def init_hash; RESOURCES.reduce({}) { |h, r| h.merge! r => {} } end
+    def set(request, response)
+      @memory[request] = response
+    end
+  end
+
+  class << self
+    BASE_URL  = 'http://pokeapi.co/api'
+    RESOURCES = {
+      'v1' => %w(
+        pokedex
+        pokemon
+        move
+        ability
+        type
+        egg
+        description
+        sprite
+        game
+      ),
+      'v2' => %w(
+        berry
+        berry-firmness
+        berry-flavor
+        contest-type
+        contest-effect
+        super-contest-effect
+        encounter-method
+        encounter-condition
+        encounter-condition-value
+        evolution-chain
+        evolution-trigger
+        generation
+        pokedex
+        version
+        version-group
+        item
+        item-attribute
+        item-category
+        item-fling-effect
+        item-pocket
+        machine
+        move
+        move-ailment
+        move-battle-style
+        move-category
+        move-damage-class
+        move-learn-method
+        move-target
+        location
+        location-area
+        pal-park-area
+        region
+        ability
+        characteristic
+        egg-group
+        gender
+        growth-rate
+        nature
+        pokeathlon-stat
+        pokemon
+        pokemon-color
+        pokemon-form
+        pokemon-habitat
+        pokemon-shape
+        pokemon-species
+        stat
+        type
+        language
+      )
+    }
+    API_VERSIONS = RESOURCES.keys
+    DEFAULT_API_VERSION = API_VERSIONS.last # Set last configured API as default
```

Comment:
I'd almost prefer `API_VERSIONS['v2']` just to be explicit, but that's nitpicky in any case.